src/components/views/room_settings/PublishedAliases.tsx

```diff
@@ -0,0 +1,403 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, {useContext, useEffect, useReducer, useRef} from 'react';
+import MatrixClientContext from "../../../contexts/MatrixClientContext";
+import { _t } from '../../../languageHandler';
+import EditableItemList from "../elements/EditableItemList";
+import OverlaySpinner from "../elements/OverlaySpinner";
+import Modal from "../../../Modal";
+import ErrorDialog from "../dialogs/ErrorDialog";
+import Field from "../elements/Field";
+import AccessibleButton from "../elements/AccessibleButton";
+import RoomPublishSetting from "./RoomPublishSetting";
+
+const UPDATE_ALT_ALIAS = 'update_alt_alias';
+const SYNCED_ALIASES = 'synced_aliases';
+const BEGIN_COMMIT = 'begin_commit';
+const END_COMMIT = 'end_commit';
+const ABORT_COMMIT = 'abort_commit';
+
+interface State {
+    alt: {
+        synced: string[],
+        working: string[],
+    },
+    canonical: {
+        synced: string,
+        working: string,
+    },
+    submitting: string | undefined,
+    regarding: string | undefined,
+    error: Error & { during?: string, errcode?: string } | undefined,
+    newAltAlias: string,
+}
+
+type Alts = string[];
+
+interface AUpdateAlias {
+    type: 'update_alt_alias',
+    value: string,
+}
+
+interface ASyncedAliases {
+    type: 'synced_aliases',
+    alt: Alts,
+    canonical: string | undefined,
+}
+
+interface ABeginCommit {
+    type: 'begin_commit',
+    submitting: 'add' | 'remove' | 'canonical',
+    alt?: Alts,
+    canonical?: string,
+    regarding?: string,
+}
+
+interface AEndCommit {
+    type: 'end_commit'
+}
+
+interface AAbortCommit {
+    type: 'abort_commit',
+    error: Error
+}
+
+type Action = AUpdateAlias | ASyncedAliases | ABeginCommit | AEndCommit | AAbortCommit;
```

Comment:
I'm a bit worried that this is a lot of extra complexity... Can we get comments as to what these are trying to achieve, and what problems they solve?

Also, our interfaces are `INamed`, so in this case would be `IUpdateAliasAction` and such. The `type Action` should probably be named something that doesn't conflict with the dispatcher, such as `type AliasAction`